compiler/rustc_typeck/src/check/check.rs

```diff
@@ -42,6 +42,17 @@ pub(super) fn check_abi(tcx: TyCtxt<'_>, span: Span, abi: Abi) {
         )
         .emit()
     }
+
+    // This ABI is only allowed on function pointers
+    if abi == Abi::CCmseNonSecureCall {
+        struct_span_err!(
+            tcx.sess,
+            span,
+            E0781,
+            "The `\"C-cmse-nonsecure-call\"` ABI is only allowed on function pointers."
```

Comment:
The message should be lowercased

```suggestion
            "the `\"C-cmse-nonsecure-call\"` ABI is only allowed on function pointers."
```